### PR TITLE
mid mod review

### PR DESCRIPTION
**Pull request by juliet-e**

### CODE REVIEW

File: app/facade/foods_facade.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class FoodsFacade
+  def self.create_food(query)
+    json = FoodsService.get_food_data(query)
+    json[:foods].map do |data|
+      Food.new(data)
+    end.take(10)
```
Comment (juliet-e) [inline app/facade/foods_facade.rb:6]: You can also limit it to 10 foods by adding the pageSize parameter to the API call. That would remove the need to add this ruby logic here

File: app/facade/foods_facade.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  end
+
+  def self.returned_items_number(query)
+    json = FoodsService.get_food_data(query)
+    json[:totalHits]
```
Comment (juliet-e) [inline app/facade/foods_facade.rb:11]: Same deal as with parks finder. You can add this to the method above and use the same API call to extract the total. Then, just return the foods and the total as an array so the controller can receive both:
`[foods, total] = FoodsFacade.create_food(params[:query])`

File: spec/features/foods/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require "rails_helper"
+
+RSpec.describe "Food Index Page", type: :feature do
+  it "displays list of attributes for food search results", :vcr do
+    food1 = Food.new(gtinupc: 124, description: "yummy food", brand_owner: "General Mills", ingredients: "hot dog, bun, mustard")
```
Comment (juliet-e) [inline spec/features/foods/index_spec.rb:5]: Typically, it's more common to pass in the data as a hash as you would expect it to come in from the API data, so all those same fields wrapped in a  { } to indicate that you're passing in a hash that's been parsed from the JSON response

File: spec/poros/food_spec.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+require "rails_helper"
+
+RSpec.describe Food do
+  it "exists and has attributes" do
+    data = {description: "Food Test Description"}
```
Comment (juliet-e) [inline spec/poros/food_spec.rb:5]: Make sure to include all the data you need in the test data, not just the description field

**Review by juliet-e — COMMENTED** [review]
Great work! I see a lot of improvement and some nice structural changes. I especially like the foundation you've laid with BaseService. I added a couple notes in both repos around refactoring to just use 1 API call. This was a tricky thing that is not strictly necessary to pass the MidMod but is a good opportunity for refactoring for cleaner, more performant code. Tests look a lot better, too. I think your next focus could be on adding more test cases for sad path and edge cases, and you'll get practice with this with consultancy. Great work, overall, Scott!
